### PR TITLE
Add implementation of SSR

### PR DESCRIPTION
**Pull request by ClaudivanFilho**
#### What is the purpose of this pull request?
- Add implementataion of SSR

#### How should this be manually tested?
[workspace link](https://claudivan--storecomponents.myvtex.com/Eletronics?O=OrderByBestDiscountDESC&Q=Smartphones&map=c%2Cc&page=1)

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/SearchResult.js [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
+export default class SearchResult extends Component {
+  static propTypes = {
+    params: PropTypes.shape({
+      /** Search's term, e.g: eletronics. */
+      term: PropTypes.string.isRequired,
+    }),
+    query: PropTypes.shape({
+      /**
+       * Rest of the search term, e.g: eletronics/smartphones/samsung implies that
+       * Q will be equal to "smartphones,samsung".
```
Comment (andreldsa) [inline react/SearchResult.js]: Q -> rest

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - `SearchFilter` compoenent
 - `SelectedFilters` component
 - Graphql queries of `search` and `facets`
+- Implementataion of SSR
```
Comment (brunojdo) [inline CHANGELOG.md]: fix typo

File: react/components/SearchResultContainer.js [hunk 3]
```diff
@@ -265,9 +259,9 @@ SearchResult.propTypes = SearchResultWithData.propTypes = {
   /** Maximum number of items per page. */
   maxItemsPerPage: PropTypes.number.isRequired,
   /** Query param. e.g: eletronics/smartphones */
-  query: PropTypes.string,
+  query: PropTypes.string.isRequired,
```
Comment (brunojdo) [inline react/components/SearchResultContainer.js]: I think we have a common denominator, a bunch of props that can be reused.

Comment (ClaudivanFilho) [reply react/components/SearchResultContainer.js]: What props?

Comment (brunojdo) [reply react/components/SearchResultContainer.js]: In SearchResult.js ```
query: PropTypes.shape({
+      /**
+       * Rest of the search term, e.g: eletronics/smartphones/samsung implies that
+       * rest will be equal to "smartphones,samsung".
+       * */
+      rest: PropTypes.string,
+      /** Determines the types of the terms, e.g: "c,c,b" (category, category, brand). */
+      map: PropTypes.string,
+      /** Search's pagination.  */
+      page: PropTypes.string,
+      /** Search's ordenation. */
+      order: PropTypes.string,
+    }),
```

**Pull request merged**